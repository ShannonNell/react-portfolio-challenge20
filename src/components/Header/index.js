import React, { useState } from 'react';
import Nav from '../../components/Nav';
import About from '../About';
import Projects from '../Portfolio';
import ContactForm from '../Contact';
import Resume from '../Resume';

function Header() {
    const [currentPage, handlePageChange] = useState('About');

    const renderPage = () => {
        // switch statement that will render appropriate selection
        switch (currentPage) {
            case "about":
                return <About></About>
            case "portfolio":
                return <Projects></Projects>
            case "contact":
                return <ContactForm></ContactForm>
            case "resume":
                return <Resume></Resume>
            default:
                return <About></About>
        }
    };
    return (
        <header className="flex-row px-1">
            <div className="flex-row">
                <h1 className="flex-row">
                    <a href="/">
                        Shannon Nell
                    </a>
                </h1>
                <div className="flex-row">
                    <Nav currentPage={currentPage} handlePageChange={handlePageChange}/>
                </div>
            </div>
            <div className="flex-row mq-pages">
                {
                    renderPage()
                }
            </div>
        </header>
    )
};

export default Header;