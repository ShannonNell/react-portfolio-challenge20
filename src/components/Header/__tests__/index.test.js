// __tests__/Nav.test.js with hard coded categories
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Header from '..';

afterEach(cleanup);

describe('Header component', () => {
  it('renders', () => {
    render(<Header></Header>);
  });

  it('matches snapshot DOM node structure', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });
});