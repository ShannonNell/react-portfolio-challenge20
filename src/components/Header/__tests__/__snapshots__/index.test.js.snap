// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <header
    class="flex-row px-1"
  >
    <h2>
      <a
        href="/"
      >
        <span
          aria-label="snell"
          role="img"
        >
          <img
            alt="snell logo"
            class="my-1"
            src="snell_logo.png"
            style="width: 50px;"
          />
        </span>
        Shannon Nell
      </a>
    </h2>
    <ul
      class="flex-row"
    >
      <li
        class="mx-1"
      >
        <a
          class="nav-link"
          href="#about"
        >
          about
        </a>
      </li>
      <li
        class="mx-1"
      >
        <a
          class="nav-link"
          href="#portfolio"
        >
          portfolio
        </a>
      </li>
      <li
        class="mx-1"
      >
        <a
          class="nav-link"
          href="#contact"
        >
          contact
        </a>
      </li>
      <li
        class="mx-1"
      >
        <a
          class="nav-link"
          href="#resume"
        >
          resume
        </a>
      </li>
    </ul>
    <div>
      <section
        class="my-5"
      >
        <h1
          id="about"
        >
          About me
        </h1>
        <img
          alt="cover"
          class="my-2"
          src="aboutme_cover.jpg"
          style="width: 100%;"
        />
        <div
          class="my-2"
        >
          <p>
            Hello, and welcome. My name is Shannon Nell and I am an upcoming full stack web developer currently enrolled in the University of Toronto's Coding Bootcamp. I have a degree in Media, Theory, Production from Western University and a diploma in Television Broadcasting as well as 3D Animation and Character Design both from Fanshawe College. I also have a diploma in Professional Photography with the Photography Institute.
          </p>
        </div>
      </section>
    </div>
  </header>
</DocumentFragment>
`;
